### ISSUE TITLE
AccountFormBase.php - checkForDuplicates sanitization & street check & default value

### ISSUE THREAD

**Issue by jellydonuts** [salesagility/SuiteCRM#1371]
The current "checkForDuplicates" function in modules/Accounts/AccountFormBase.php is not really the greatest... 
1. It's missing even the most basic Sanitization of the address fields (such as space characters, etc).
2. Also, it's missing a duplicate check for the street field.
3. The duplicate check only checks to see if the field is not empty, however if you've assigned default values to these fields, the duplicate check will not work properly

Proposed fix: 
1. At least add trim() around the variables being placed into the SQL query
2. Add billing_address_street & shipping_address_street into the search query
`$temp_query =  "  billing_address_street like '".trim($_POST[$prefix.'billing_address_street'])."%'";`
3. replace the if(!empty($_POST... with if $_POST field does not equal default value of field.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [AccountFormBase.php - checkForDuplicates sanitization & street check & default value](https://trello.com/c/VsdPMIxS/98-accountformbase-php-checkforduplicates-sanitization-street-check-default-value)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
